backend/src/embedding/openai-embbeding-provider.ts

```
@@ -3,29 +3,41 @@
-  private static logger = new Logger(openAIEmbProvider.name);
-
-  private static openai = () => {
-    return new OpenAI({
-      apiKey: process.env.OPEN_API_KEY,
+export class OpenAIEmbProvider {
+  private logger = new Logger(OpenAIEmbProvider.name);
+  private static instance: OpenAIEmbProvider;
+  private openAi: OpenAI;
+  static openAi: any;
+  static getInstance() {
+    if (this.instance) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)